From b076179082dc4b91b5f8de07fe0b61c4131f7881 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Mon, 3 Oct 2022 23:05:14 +0200
Subject: [PATCH] nss-dp: adapt to netif_napi_add() changes

netif_napi_add() removed the weight argument and just uses the default
NAPI_POLL_WEIGHT in background, so for those requiring custom weight use
netif_napi_add_weight() instead.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 hal/dp_ops/edma_dp/edma_v1/edma_data_plane.c | 3 +--
 hal/dp_ops/edma_dp/edma_v2/edma_cfg_rx.c     | 4 ++--
 hal/dp_ops/edma_dp/edma_v2/edma_cfg_tx.c     | 4 ++--
 hal/dp_ops/syn_gmac_dp/syn_dp.c              | 4 ++--
 4 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/hal/dp_ops/edma_dp/edma_v1/edma_data_plane.c b/hal/dp_ops/edma_dp/edma_v1/edma_data_plane.c
index 7008c94..5ab6bc6 100644
--- a/hal/dp_ops/edma_dp/edma_v1/edma_data_plane.c
+++ b/hal/dp_ops/edma_dp/edma_v1/edma_data_plane.c
@@ -836,8 +836,7 @@ static int edma_register_netdevice(struct net_device *netdev, uint32_t macid)
 	 * NAPI add
 	 */
 	if (!edma_hw.napi_added) {
-		netif_napi_add(netdev, &edma_hw.napi, edma_napi,
-				NAPI_POLL_WEIGHT);
+		netif_napi_add(netdev, &edma_hw.napi, edma_napi);
 		/*
 		 * Register the interrupt handlers and enable interrupts
 		 */
diff --git a/hal/dp_ops/edma_dp/edma_v2/edma_cfg_rx.c b/hal/dp_ops/edma_dp/edma_v2/edma_cfg_rx.c
index e2bb97d..d49728b 100644
--- a/hal/dp_ops/edma_dp/edma_v2/edma_cfg_rx.c
+++ b/hal/dp_ops/edma_dp/edma_v2/edma_cfg_rx.c
@@ -1097,8 +1097,8 @@ void edma_cfg_rx_napi_add(struct edma_gbl_ctx *egc, struct net_device *netdev)
 
 	for (i = 0; i < egc->num_rxdesc_rings; i++) {
 		struct edma_rxdesc_ring *rxdesc_ring = &egc->rxdesc_rings[i];
-		netif_napi_add(netdev, &rxdesc_ring->napi,
-				edma_rx_napi_poll, nss_dp_rx_napi_budget);
+		netif_napi_add_weight(netdev, &rxdesc_ring->napi,
+				      edma_rx_napi_poll, nss_dp_rx_napi_budget);
 		rxdesc_ring->napi_added = true;
 	}
 	edma_info("%s: Rx NAPI budget: %d\n", netdev->name, nss_dp_rx_napi_budget);
diff --git a/hal/dp_ops/edma_dp/edma_v2/edma_cfg_tx.c b/hal/dp_ops/edma_dp/edma_v2/edma_cfg_tx.c
index 4f9e298..fba224f 100644
--- a/hal/dp_ops/edma_dp/edma_v2/edma_cfg_tx.c
+++ b/hal/dp_ops/edma_dp/edma_v2/edma_cfg_tx.c
@@ -672,8 +672,8 @@ void edma_cfg_tx_napi_add(struct edma_gbl_ctx *egc, struct net_device *netdev)
 	for (i = 0; i < egc->num_txcmpl_rings; i++) {
 		struct edma_txcmpl_ring *txcmpl_ring = &egc->txcmpl_rings[i];
 
-		netif_napi_add(netdev, &txcmpl_ring->napi,
-				edma_tx_napi_poll, nss_dp_tx_napi_budget);
+		netif_napi_add_weight(netdev, &txcmpl_ring->napi,
+				      edma_tx_napi_poll, nss_dp_tx_napi_budget);
 		txcmpl_ring->napi_added = true;
 	}
 	edma_info("Tx NAPI budget: %d\n", nss_dp_tx_napi_budget);
diff --git a/hal/dp_ops/syn_gmac_dp/syn_dp.c b/hal/dp_ops/syn_gmac_dp/syn_dp.c
index a3032b4..2f79fb9 100644
--- a/hal/dp_ops/syn_gmac_dp/syn_dp.c
+++ b/hal/dp_ops/syn_gmac_dp/syn_dp.c
@@ -189,8 +189,8 @@ static int syn_dp_if_init(struct nss_dp_data_plane_ctx *dpc)
 	}
 
 	if (!dev_info->napi_added) {
-		netif_napi_add(netdev, &rx_info->napi_rx, syn_dp_napi_poll_rx, SYN_DP_NAPI_BUDGET_RX);
-		netif_napi_add(netdev, &tx_info->napi_tx, syn_dp_napi_poll_tx, SYN_DP_NAPI_BUDGET_TX);
+		netif_napi_add_weight(netdev, &rx_info->napi_rx, syn_dp_napi_poll_rx, SYN_DP_NAPI_BUDGET_RX);
+		netif_napi_add_weight(netdev, &tx_info->napi_tx, syn_dp_napi_poll_tx, SYN_DP_NAPI_BUDGET_TX);
 
 		/*
 		 * Requesting irq. Set IRQ_DISABLE_UNLAZY flag, this flag
-- 
2.37.3

